cmake/modules/hunter_add_host_subdirectory.cmake

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2016 Fabian Winnen
+# All rights reserved.
+
+# Adds a subdirectory with CMakeList.txt as seperate build with default (host) toolchain
+
+# Usage:
+#   hunter_add_host_subdirectory(host)
+#   hunter_add_host_subdirectory(host
+#       BINARY_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/host-build
+#       ENV_VARS CC CXX
+#   )
+
+# Note:
+#   With polly Android toolchain this fails if you do not unset CC and CXX ENV{} variables.
+#   For this and possible other dependent ENV variables you can ue ENV_VARS.
+#   Every entry in ENV_VARS will get saved and unset for the host project and restored
+#   after the host project build finished
+
+include(CMakeParseArguments) # cmake_parse_arguments
+
+include(hunter_internal_error)
+include(hunter_status_print)
+include(hunter_status_debug)
+
+# internal variables: _hunter_ahs_*
+macro(hunter_add_host_subdirectory)
```

Comment:
Do we need to use `macro` here? Can you rewrite it in terms of `function`? In this case we can drop all this `_hunter_ahs_*` variables.